recipe/meta.yaml

```diff
@@ -43,18 +50,29 @@ test:
     - libpysal.io.tests
     - libpysal.io.util
     - libpysal.weights
-# pip check is not smart enough to know that an import name is not the package name :-(
-# commands:
-#  - pip check
-# requires:
-#  - pip
+  commands:
+    - pip check
+      # smoke test on omly test weight due to the rest of the test using all relative imports to the source code.
+      # skipping test_plot due to reading a datafile, importing matplotlib and warnings causing overall test to fail.
+    - pytest libpysal/weights/tests/test_weights.py -k "not test_plot"
```

Comment:
I believe you could run something like `pytest --pyargs libpysal -vv`
You could add in require: matplotlib-base, pytest-xdist https://github.com/pysal/libpysal/blob/v4.10/pyproject.toml#L69-L77
That being said, most would not run because of either missing test dependencies or test data. 
You can see where tests data is downloaded here: https://github.com/pysal/libpysal/blob/v4.10/.github/workflows/unittests.yml#L62-L77

I am okay with this smoke test.

Reply:
Unfortunately, all the tests have relative imports pointing to the source code. We could patch them all, but given the time constraints, having a smoke test is enough.